packages/liveblocks-core/src/protocol/AuthToken.ts

```diff
@@ -22,7 +22,9 @@ export type RoomAuthToken = {
 
   // Extra payload as defined by the customer's own authorization
   id?: string;
+  anonymousId?: string;
```

Comment:
Will either  `id` or  `anonymousId` always be set, and the other always be empty (like an XOR)? If so, it'd be helpful to codify that knowledge right in the type definition with something like:

```ts
export type RoomAuthToken = {
  ...
} & (
  | { id: string; anonymousId?: never }
  | { id?: never; anonymousId: string }
);
```

That way, TypeScript can be aware of the relation between these fields, and for example help us understand that if one is missing, the other field will be present.

Reply:
Yes it's a XOR, I updated the type. thank you!